File: 1588. Sum of All Odd Length Subarrays/Solution.py
from typing import List


class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        n = len(arr)
        return sum(((i + 1) * (n - i) + 1) // 2 * arr[i] for i in range(n))

File: 1588. Sum of All Odd Length Subarrays/Solution.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int res = 0;
        for (int i = 0; i < arr.length; i++) {
            res += ((i + 1) * (arr.length - i) + 1) / 2 * arr[i];
        }
        return res;
    }
}
